src/app/data-entry-multi-form/data-entry-multi-form.component.ts

```diff
@@ -66,39 +70,50 @@ export class DataEntryMultiFormComponent implements OnInit {
         }
       });
     }
+    this._dataEntryStepsDef = this.getFlatElements()
   }
 
   onReviewAndSubmit() {
     console.log('Review and Submit');
   }
 
-  getField(id, steps, isChild = false) {
-
+  getField(id, steps) {
     steps.forEach((step, index) => {
-      // this._currentStepIndex = index;
       if (step.id === id) {
-        if (!isChild) {
-          this._currentStepIndex = index;
-        }
-        this._currentChildStepIndex = isChild ? index : -1;
-
         this._currentStep = step;
-
       } else {
-
         if (step.steps?.length > 0) {
-          this.getField(id, step.steps, true)
+          this.getField(id, step.steps)
         }
-
       }
     });
-    console.log(this._currentStepIndex, 'current')
-    console.log(this._currentChildStepIndex, 'child')
   }
-  onPanelChange(id: string) {
-    // this._currentStepIndex = this.dataEntryForm.steps.findIndex(step => step.id === id);
 
-    this.getField(id, this.dataEntryForm.steps, false)
+  public getFlatElements() {
+    const results = this.dataEntryForm.steps;
+    const flat: DataEntrySteps[] = [];
+    const flatten = (array: any) => {
+      for (let i in array) {
+        const item = array[i];
+        flat.push({
+          id: item.id,
+          text: item.label
+        });
```

Comment:
Do we need to truncate data? Why not just push the item itself in?

Reply:
I need only label and text for the stepper, don't need all of the object